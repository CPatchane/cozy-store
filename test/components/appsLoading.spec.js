'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import {
  AppsLoading,
  LoadingAppsComponents,
  Placeholder
} from 'ducks/components/AppsLoading'

global.Math.random = () => 1 // remove random for testing

describe('AppsLoading component', () => {
  it('should be rendered correctly and never be updated', () => {
    const wrapper = shallow(<AppsLoading />)
    const shouldUpdate = wrapper.instance().shouldComponentUpdate()
    expect(shouldUpdate).toBe(false)
  })
})

describe('LoadingAppsComponents component', () => {
  it('should be rendered correctly', () => {
    const component = shallow(
      <LoadingAppsComponents count={3} subKey="mock" />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should be rendered correctly if isMobile', () => {
    const component = shallow(
      <LoadingAppsComponents
        count={3}
        subKey="mock"
        breakpoints={{ isMobile: true }}
      />
    ).getElement()
    expect(component).toMatchSnapshot()
  })
})

describe('Placeholder component', () => {
  it('should be rendered correctly', () => {
    const component = shallow(
      <Placeholder width="5rem" height="3rem" />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should be rendered correctly with autoMargin option', () => {
    const component = shallow(
      <Placeholder width="5rem" height="3rem" autoMargin />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should be rendered correctly with interval of widths', () => {
    const component = shallow(
      <Placeholder width={[3, 5]} height="3rem" autoMargin />
    ).getElement()
    expect(component).toMatchSnapshot()
  })
})
