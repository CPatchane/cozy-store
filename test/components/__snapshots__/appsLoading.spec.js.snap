// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingAppsComponents component should be rendered correctly 1`] = `
<div
  className="sto-sections-list"
>
  <div
    className="sto-small-app-item"
  >
    <div
      className="sto-small-app-item-icon-wrapper"
    >
      <div
        className="sto-small-app-item-icon"
      >
        <Placeholder
          autoMargin={true}
          height="3rem"
          width="3rem"
        />
      </div>
    </div>
    <div
      className="sto-small-app-item-desc"
    >
      <h4
        className="sto-small-app-item-title"
      >
        <Placeholder
          autoMargin={true}
          height="1.1rem"
          width={
            Array [
              4,
              8,
            ]
          }
        />
      </h4>
      <p
        className="sto-small-app-item-developer"
      >
        <Placeholder
          autoMargin={true}
          height=".8rem"
          width={
            Array [
              4,
              8,
            ]
          }
        />
      </p>
      <p
        className="sto-small-app-item-status"
      >
        <Placeholder
          autoMargin={true}
          height=".8rem"
          width="4rem"
        />
      </p>
    </div>
  </div>
  <div
    className="sto-small-app-item"
  >
    <div
      className="sto-small-app-item-icon-wrapper"
    >
      <div
        className="sto-small-app-item-icon"
      >
        <Placeholder
          autoMargin={true}
          height="3rem"
          width="3rem"
        />
      </div>
    </div>
    <div
      className="sto-small-app-item-desc"
    >
      <h4
        className="sto-small-app-item-title"
      >
        <Placeholder
          autoMargin={true}
          height="1.1rem"
          width={
            Array [
              4,
              8,
            ]
          }
        />
      </h4>
      <p
        className="sto-small-app-item-developer"
      >
        <Placeholder
          autoMargin={true}
          height=".8rem"
          width={
            Array [
              4,
              8,
            ]
          }
        />
      </p>
      <p
        className="sto-small-app-item-status"
      >
        <Placeholder
          autoMargin={true}
          height=".8rem"
          width="4rem"
        />
      </p>
    </div>
  </div>
  <div
    className="sto-small-app-item"
  >
    <div
      className="sto-small-app-item-icon-wrapper"
    >
      <div
        className="sto-small-app-item-icon"
      >
        <Placeholder
          autoMargin={true}
          height="3rem"
          width="3rem"
        />
      </div>
    </div>
    <div
      className="sto-small-app-item-desc"
    >
      <h4
        className="sto-small-app-item-title"
      >
        <Placeholder
          autoMargin={true}
          height="1.1rem"
          width={
            Array [
              4,
              8,
            ]
          }
        />
      </h4>
      <p
        className="sto-small-app-item-developer"
      >
        <Placeholder
          autoMargin={true}
          height=".8rem"
          width={
            Array [
              4,
              8,
            ]
          }
        />
      </p>
      <p
        className="sto-small-app-item-status"
      >
        <Placeholder
          autoMargin={true}
          height=".8rem"
          width="4rem"
        />
      </p>
    </div>
  </div>
</div>
`;

exports[`LoadingAppsComponents component should be rendered correctly if isMobile 1`] = `
<div
  className="sto-sections-list"
>
  <div
    className="sto-small-app-item"
  >
    <div
      className="sto-small-app-item-icon-wrapper"
    >
      <div
        className="sto-small-app-item-icon"
      >
        <Placeholder
          autoMargin={true}
          height="2.5rem"
          width="2.5rem"
        />
      </div>
    </div>
    <div
      className="sto-small-app-item-desc"
    >
      <h4
        className="sto-small-app-item-title"
      >
        <Placeholder
          autoMargin={true}
          height=".8rem"
          width={
            Array [
              4,
              5,
            ]
          }
        />
      </h4>
      <p
        className="sto-small-app-item-developer"
      >
        <Placeholder
          autoMargin={true}
          height=".8rem"
          width={
            Array [
              4,
              5,
            ]
          }
        />
      </p>
      <p
        className="sto-small-app-item-status"
      >
        <Placeholder
          autoMargin={true}
          height=".6rem"
          width="4rem"
        />
      </p>
    </div>
  </div>
  <div
    className="sto-small-app-item"
  >
    <div
      className="sto-small-app-item-icon-wrapper"
    >
      <div
        className="sto-small-app-item-icon"
      >
        <Placeholder
          autoMargin={true}
          height="2.5rem"
          width="2.5rem"
        />
      </div>
    </div>
    <div
      className="sto-small-app-item-desc"
    >
      <h4
        className="sto-small-app-item-title"
      >
        <Placeholder
          autoMargin={true}
          height=".8rem"
          width={
            Array [
              4,
              5,
            ]
          }
        />
      </h4>
      <p
        className="sto-small-app-item-developer"
      >
        <Placeholder
          autoMargin={true}
          height=".8rem"
          width={
            Array [
              4,
              5,
            ]
          }
        />
      </p>
      <p
        className="sto-small-app-item-status"
      >
        <Placeholder
          autoMargin={true}
          height=".6rem"
          width="4rem"
        />
      </p>
    </div>
  </div>
  <div
    className="sto-small-app-item"
  >
    <div
      className="sto-small-app-item-icon-wrapper"
    >
      <div
        className="sto-small-app-item-icon"
      >
        <Placeholder
          autoMargin={true}
          height="2.5rem"
          width="2.5rem"
        />
      </div>
    </div>
    <div
      className="sto-small-app-item-desc"
    >
      <h4
        className="sto-small-app-item-title"
      >
        <Placeholder
          autoMargin={true}
          height=".8rem"
          width={
            Array [
              4,
              5,
            ]
          }
        />
      </h4>
      <p
        className="sto-small-app-item-developer"
      >
        <Placeholder
          autoMargin={true}
          height=".8rem"
          width={
            Array [
              4,
              5,
            ]
          }
        />
      </p>
      <p
        className="sto-small-app-item-status"
      >
        <Placeholder
          autoMargin={true}
          height=".6rem"
          width="4rem"
        />
      </p>
    </div>
  </div>
</div>
`;
